components/brave_wallet/browser/blockchain_registry.cc

```diff
@@ -274,4 +279,31 @@ void BlockchainRegistry::GetTopDapps(const std::string& chain_id,
   std::move(callback).Run(std::move(dapps_copy));
 }
 
+std::string BlockchainRegistry::GetCoingeckoId(
+    const std::string& chain_id,
+    const std::string& contract_address) {
+  const auto& chain_id_lower = base::ToLowerASCII(chain_id);
+  if (!coingecko_ids_map_.contains(chain_id_lower)) {
+    return "";
+  }
+
+  if (coingecko_ids_map_[chain_id_lower].contains(contract_address)) {
+    return coingecko_ids_map_[chain_id_lower][contract_address];
+  }
+
+  if (coingecko_ids_map_[chain_id_lower].contains(
+          base::ToLowerASCII(contract_address))) {
+    return coingecko_ids_map_[chain_id_lower]
+                             [base::ToLowerASCII(contract_address)];
+  }
+
+  return "";
+}
```

Comment:
Let's avoid map of maps and use pair as a key:
`using CoingeckoIdsMap = base::flat_map<std::pair<std::string, std::string>, std::string>;`
Also let's `base::ToLowerASCII` both parts of pair when creating map

Reply:
https://github.com/brave/brave-core/pull/19489/commits/0456fc4a6ddca7a0be77d8aee8d30b6a0508658d